### PR TITLE
Add libzopfli

### PR DESCRIPTION
**Pull request by hmaarrfk**
Needed for imagecodecs

Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ ] Source is from official source
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by hmaarrfk** [comment]
@conda-forge/help-c-cpp any idea why VC2008 fails on azure? will this be a problem when the feedstock rerenders?

File: recipes/libzopfli/bld.bat [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    -DCMAKE_INSTALL_PREFIX=%LIBRARY_PREFIX%                               ^
+    -DCMAKE_PREFIX_PATH=%LIBRARY_PREFIX%                                  ^
+    -DCMAKE_INSTALL_INCLUDEDIR=%LIBRARY_INC%\zopfli                       ^
+    -DBUILD_SHARED_LIBS=1                                                 ^
+    -DZOPFLI_BUILD_SHARED=1                                               ^
+    ..
+
+if errorlevel 1 exit 1
+
+cmake --build . --target install --config Release
```
Comment (marcelotrevisani) [inline recipes/libzopfli/bld.bat]: just 
```suggestion
ninja install
```
is enough?

File: recipes/libzopfli/conda_build_config.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+c_compiler:                    # [win]
```
Comment (marcelotrevisani) [inline recipes/libzopfli/conda_build_config.yaml]: why are you overwriting these keys?

Comment (hmaarrfk) [reply recipes/libzopfli/conda_build_config.yaml]: because i need visual studio 2017.

**Comment by hmaarrfk** [comment]
i'll have to build this locally some time.

**Comment by hmaarrfk** [comment]
@marcelotrevisani crossing my fingers that this finally did it!

File: recipes/libzopfli/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    # https://github.com/google/zopfli/pull/164
+    - include_more_headers.patch
+
+build:
+  number: 0
+  # Compiling with VS2008 works, but it seems to have
+  # libzopfli.lib/dll instead of
+  # zopfli.lib/dll
+  # And I don't really have patience to support python 2.7 given that it is 2020.
+  skip: True  # [win and vc==9]
```
Comment (xhochy) [inline recipes/libzopfli/meta.yaml:21]: We (conda-forge) neither, these builds are now disabled by default.
```suggestion
```

**Comment by xhochy** [comment]
The `skip` shouldn't be needed, otherwise this looks ready for a merge.

**Comment by hmaarrfk** [comment]
Thanks @xhochy

**Pull request merged**